tests/clone.rs

```diff
@@ -170,3 +171,27 @@ mod import_rename {
         closure(0, 0);
     }
 }
+
+#[test]
+fn derive_downgrade() {
+    #[derive(Downgrade)]
+    pub struct NewType(Object);
```

Comment:
What's the advantage of the macro over putting your struct into an `Rc`/`Arc`?

Reply:
Easier to add methods to your structure.

Reply:
But more expensive as each field has to be handled separately. But for the `NewType` pattern above I would use it quite often personally, the others should probably get a mention about the performance impact in the documentation :)

Reply:
I also started with `NewType` pattern with single gtk widget in it. Later I found that I want a `struct MyListView<T: ToString>(gtk::Widget, PhantomData<T>)`. So, I decided to generalize macro.